soc/intel/intel_ish/Kconfig

```diff
@@ -13,3 +13,4 @@ config SOC_FAMILY_INTEL_ISH
 	select INTEL_HAL
 	select HAS_PM
 	select HAS_COVERAGE_SUPPORT
+	select DYNAMIC_INTERRUPTS
```

Comment:
This seems unrelated - is this really needed?

Reply:
It's required by APIC timer driver drivers/timer/apic_tsc.c who calls irq_connect_dynamic() function.

Reply:
Not sure how is the normal practice here, but shouldn't it be defined at `drivers/timer/Kconfig.x86` then?

Reply:
all existing x86 CPUs select DYNAMIC_INTERRUPTS in SoC Kconfig, so I just follow. Maybe they have other reason to select, not only for APIC timer driver.
I uploaded a new patch to add dependency of DYNAMIC_INTERRUPTS for APIC TSC timer, https://github.com/zephyrproject-rtos/zephyr/pull/74127.
In addition, for ISH CPUs, select DYNAMIC_INTERRUPTS only when APIC_TIMER_TSC is set.

Reply:
I guess that making it clear on the driver that it depends on the dynamic interrupts is useful to figure out the reason for the selection here. On a side note, looking into dynamic interrupts and x86, things seem a bit messy - guess I'll add some things on my TODO list...